framework/src/meshgenerators/BlockDeletionGenerator.C

```diff
@@ -21,19 +20,39 @@ BlockDeletionGenerator::validParams()
   InputParameters params = ElementDeletionGeneratorBase::validParams();
 
   params.addClassDescription(
-      "Mesh generator which removes elements with the specified subdomain ID");
-  params.addRequiredParam<SubdomainID>("block_id", "The block to be deleted");
+      "Mesh generator which removes elements from the specified subdomains");
+  params.addParam<std::vector<SubdomainName>>("block", "The list of blocks to be deleted");
+  params.addDeprecatedParam<SubdomainID>(
+      "block_id", "The block to be deleted.", "Use block instead");
 
   return params;
 }
 
 BlockDeletionGenerator::BlockDeletionGenerator(const InputParameters & parameters)
-  : ElementDeletionGeneratorBase(parameters), _block_id(getParam<SubdomainID>("block_id"))
+  : ElementDeletionGeneratorBase(parameters),
+    _block_names(getParam<std::vector<SubdomainName>>("block"))
 {
+  // Handle deprecated parameter
+  if (isParamValid("block_id"))
+    _block_ids.push_back(getParam<SubdomainID>("block_id"));
+  if (!isParamValid("block_id") && !isParamValid("block"))
+    mooseError("The list of blocks to be deleted should be provided.");
+  if (isParamValid("block_id") && isParamValid("block"))
+    mooseError("Please do not provide both block and block_id to the BlockDeletionGenerator.");
```

Comment:
```suggestion
    paramError("block_id", "Cannot use with the parameter 'block'. Please use just 'block'.");
```

Or something of the sort